### PR TITLE
Add guideline pages

### LINKED ISSUE #42: Add guideline pages
### The index page `/guidelines`

Can have guideline category pages in the following format:

In a card form with the following contents:
 * Image
 * Title
 * Description

#### Inspirations
 * https://www.intercom.com/help/en/

### Guideline category pages `/guidelines/:category`

Can have a list of the guidelines in the category.

Should have its attributes that're seen on the parent page visible as well. Such as:
 * Image
 * Title
 * Description

#### Inspirations
 * https://www.intercom.com/help/en/collections/2094785-providing-help-content-with-intercom-articles
 * https://getbootstrap.com/docs/4.0/components/breadcrumb/

### Guideline pages `/guidelines/:category/:title`

Should render a markdown.

#### Inspirations
 * https://www.intercom.com/help/en/articles/56645-how-to-write-great-help-articles

## Routes
The following routes should be reachable:

 * `/guidelines`
 * `/guidelines/:category`
 * `/guidelines/:category/:title`

### `:category`
One of the following:
 * common
 * aid-crews
 * elderly-crews
 * coordination-crews
 * triage-crews
 * hospital-crews
 * research-ambassadors

#### Notes
For images, the following graphics could be used.

Common: `undraw_reading_0re1.svg` in https://www.notion.so/Assets-04b473617ece497291aa7fbff4b8d912
Others: The same images used on the landing page.

### `:title`
URL should be in hyphen-separated case aka kebab-case.

### Example route
/guidelines/elderly-crews/role-definition

## Design considerations
* Could use the limites set of typography options used in other pages. The naming of these classes could be improved.

## Linking
In the top bar, we could remove the FAQ link and put a "Guidelines" link instead.

### PR DESCRIPTION
**Pull request by dimanyc**
Closes #42

This PR handles adding guideline resources are described in #42.

To follow the requested URL structure (comment: https://github.com/panvol/pandemic-volunteers/issues/42#issuecomment-615248137), an default locale of `en` is introduced and used as a scope in routes, to ensure that `/en/guidelines` and `/guidelines` route default to the same resource

Because `guidelines` are ephemeral, a presenter is used to display initial guidelines that have guideline categories that, in turn, have many articles. Presenters also take care of fetching and rendering metadata stored in YML files

### Dimitry's TODO:
- [x] add markdown parsing / rendering based on https://github.com/panvol/pandemic-volunteers/pull/44/files#diff-e55a6c615d18aeb75e0610b835b16658
- [x] make methods such as `GuidelineCategory.extract_slug` etc., ~~metadata
  file fetching and loading~~(done), and loading dirs via `Pathname` reusable.
  This is because GuidelineCategory and GuidelineArticle will be sharing
  90% of their methods. GuidelineArticle will also need to be able to
  load its own metadata
- [x] move `GuidelinesPresenter` to `app/models` since `GuidelineCategory`
  is very similar and acts as a stand-in for a model
- [x] add designs

### Han's TODO: 
 * [x] Breadcrumbs
    * https://getbootstrap.com/docs/4.0/components/breadcrumb/
    * https://docs.pandemicvolunteers.org/en/articles/3924739-example-article
 * [x] Refactor
    * [x] Remove models and services & Use views and helpers
    * [x] Work on guideline view file naming and directories (Along with common and shared)
    * [x] Consistency among Pages and Guidelines controllers and flows
 * [x] Cache file system and renderings
    * We could use md5 hashes for detecting changes
    * This could allow us to remove the initializer and partials for PagesControllers
 * [ ] Add [real guidelines](https://docs.google.com/document/d/1v2SoFNTEw5eTEXr8bsdSGlyuZfCBpM33fgi0sVFUjng)

### CODE REVIEW

File: app/controllers/guideline_categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class GuidelineCategoriesController < ApplicationController
+  def index
+    @category =
```
Comment (dimanyc) [inline app/controllers/guideline_categories_controller.rb:3]: this is a WIP

**Comment by hantuzun** [comment]
Hi @ozkc -- We only support one language at the moment. But I've referenced your comment from issue #4

**Comment by hantuzun** [comment]
# Current State

## Guidelines Index
Screen Shot 2020-04-20 at 01 03 14

## Category pages
Screen Shot 2020-04-20 at 01 03 18

## Article pages
Screen Shot 2020-04-20 at 01 03 22

# TODO
 * [ ] Breadcrumbs
    * https://getbootstrap.com/docs/4.0/components/breadcrumb/
    * https://docs.pandemicvolunteers.org/en/articles/3924739-example-article
 * [ ] Refactor
    * [ ] Remove models and services & Use views and helpers
    * [ ] Work on guideline view file naming and directories (Along with common and shared)
    * [ ] Consistency pages and guidelines
 * [ ] Cache file system and renderings
    * We could use md5 hashes for detecting changes
    * This could allow us to remove the initializer and partials for PagesControllers
 * [ ] Add [real guidelines](https://docs.google.com/document/d/1v2SoFNTEw5eTEXr8bsdSGlyuZfCBpM33fgi0sVFUjng)

**Comment by dimanyc** [comment]
pages look great! gj @hantuzun

**Comment by hantuzun** [comment]
Looks good! @dimanyc

**Pull request closed without merging**